File: clothes/sol.js
const readline = require('readline').createInterface({
  input: process.stdin,
})

const solve = (n, m, prices, matches) => {
  if (n < 3 || m < 3) {
    return -1
  }
  const matching_costs = Array.from({ length: n }, () =>
    Array.from({ length: n }, () => Infinity)
  )
  for (let i = 0; i < m; i++) {
    const [a, b] = matches[i].split(' ').map(Number)
    matching_costs[a - 1][b - 1] = matching_costs[b - 1][a - 1] = Math.min(
      prices[a - 1],
      prices[b - 1]
    )
  }
  let min_cost = Infinity
  for (let i = 0; i < n; i++) {
    for (let j = i + 1; j < n; j++) {
      if (matching_costs[i][j] == Infinity) {
        continue
      }
      for (let k = j + 1; k < n; k++) {
        if (
          matching_costs[i][k] != Infinity &&
          matching_costs[j][k] != Infinity
        ) {
          min_cost = Math.min(min_cost, prices[i] + prices[j] + prices[k])
        }
      }
    }
  }
  if (min_cost == Infinity) {
    return -1
  } else {
    return min_cost
  }
}

const main = async () => {
  const [n, m] = (await new Promise((resolve) => rl.question('', resolve))).split(' ').map(Number)
  const prices = (await new Promise((resolve) => rl.question('', resolve))).split(' ').map(Number)
  const matches = []

  for (let i = 0; i < m; i++) {
    const input = (await new Promise((resolve) => rl.question('', resolve)))
    matches.push(input)
  }
  
  rl.close()
  console.log(solve(n, m, prices, matches))
}




main()

File: clothes/sol.py
def optimized_Sollution(input_file):
    n, m = map(int, input().split())
    if (n < 3) or (m<3): return -1 
    prices = list(map(int, input().split()))
    matching_costs = [[float('inf') for _ in range(n)] for _ in range(n)]
    for _ in range(m):
        i, j = map(int, input().split())
        matching_costs[i-1][j-1] = matching_costs[j-1][i-1] = min(prices[i-1], prices[j-1])

        
    min_cost = float('inf')
    for i in range(n):
        for j in range(i+1, n):
            if matching_costs[i][j] == float('inf'):
                continue
            for k in range(j+1, n):
                if matching_costs[i][k] != float('inf') and matching_costs[j][k] != float('inf'):
                    min_cost = min(min_cost, prices[i]+prices[j]+prices[k])

    if min_cost == float('inf'):
        return -1
    else:
        return min_cost
print(optimized_Sollution("input.txt"))